editoast/src/views/infra/auto_fixes/mod.rs

```diff
@@ -44,6 +44,30 @@ fn new_ref_fix_create_pair(object: RailjsonObject) -> (ObjectRef, Fix) {
     (object_ref, (operation, cache_operation))
 }
 
+mod ordered_operation {
+    #![allow(clippy::non_canonical_partial_ord_impl)]
+    // ALLOW: `PartialOrd` and `Ord` are defined exactly the same with `derivative`
+    // Because of `derive` macro, applying `#[allow()]` on the `struct` doesn't
+    // work. Therefore, isolating in a submodule and allowing for the whole
+    // module.
+    use crate::schema::ObjectRef;
+
+    #[derive(Debug, Clone, PartialEq, Eq, derivative::Derivative)]
+    #[derivative(
+        PartialOrd = "feature_allow_slow_enum",
+        Ord = "feature_allow_slow_enum"
+    )]
+    pub enum OrderedOperation {
+        RemoveTrackRange {
+            track_range_idx: usize,
+            #[derivative(PartialOrd = "ignore", Ord = "ignore")]
+            object_ref: ObjectRef,
+        },
+        Delete(#[derivative(PartialOrd = "ignore", Ord = "ignore")] ObjectRef),
```

Comment:
I don't see why:
- We would need `Delete` to be an ordered operation. 
- We need to keep the `object_ref` context

Reply:
`ObjectRef` is indeed useless, thanks for noticing because it does make a whole lot of weird code going away! Done in https://github.com/osrd-project/osrd/pull/6012/commits/b55066ffd3fc8dadd4a72120f03cfeb82b171d6f.

For the reason why we would need the `Delete`, I chose to handle all intermediate `Operation` as the same, mostly to avoid other weirder (subjective) code. This other solution would mostly means something along these lines (pseudo-code).

```
let (invalid_references, others) = infra_errors.partition(|e| get_invalid_reference_separated_from_the_rest_of_errors(e.get_sub_type());

invalid_references
    .into_iter()
    .map(|e| match e.get_sub_type() {
        InfraError::InvalidReference { reference } => get_track_index(reference),
        _ => unreachable!("we only have InvalidReference in here"), // <<--- this is kind of weird
    })
    .map(OrderedOperation::from)
    .sort_by_key(...)
    .reduce(reduce_operation)
    .collect();

// Then do something similar on 'others' which will also have some unreachable patterns.

// Then reduce_operation between the results of handling 'invalid_references' and 'others'
```

If you believe it's better, I can try to write that to see in the details if does really work and if it's a good alternative.

Reply:
I see. An idea could be to make `OrderedOperation` a wrapper around `Operation`.

```rust
struct OrderedOperation{
  operation: Operation,
  order: u64,
}
```

The approach is to order all operations without distinction.

Reply:
After discussion, the proposition doesn't add much, the fundamental problem being that JSON Patch lost some semantic (we want to remove some specific track ID, not an index). The code is not entirely satisfactory as it is, but no version would be without addressing the more fundamental problem.

An idea would be to modify how the RailJSON objects are expressed. For example, if `track_ranges` would be a map, then the JSONPatch would be able to keep the intended semantic, and ordering operations would not be needed anymore. It's possible such refactor happen in the future (other problems seems to point in that direction too).